### DISCUSSION TITLE
Does anyone know any JAR library that uses Telnet?

### DISCUSSION THREAD

**Discussion by Slicudis** [logisim-evolution/logisim-evolution#2029] [Q&A] [answered]
Does anyone know if there's a library that uses the same system that the program "Digital" uses to connect PuTTY to your circuit instead of using the TTY and Keyboard components?

**Comment by maehne** [comment]
I am not aware of such a JAR library for Logisim-evolution. You probably already could implement that behaviour using the `TCL generic` block from the TCL library. However, that component is only partially documented, requires co-simulation with ModelSim/Questa Sim, and hasn't been used a lot in the past years. Implementing it as a dedicated component is certainly feasible with reasonable effort. Pull requests in that direction are welcome!

**Comment by zdimension** [comment, marked as answer]
This component was added in #2104. It matches the one in Digital (which I used as direct inspiration).

![image](https://github.com/user-attachments/assets/6645928b-12dd-444d-9e26-45178fff1597)
